modern-business/inc/fse-template-data.php

```diff
@@ -0,0 +1,324 @@
+<?php
+/**
+ * Template data inserter file.
+ *
+ * @package full-site-editing
+ */
+
+/**
+ * Class A8C_WP_Template_Data_Inserter
+ */
+class A8C_WP_Template_Data_Inserter {
+	/**
+	 * This function will be called on plugin activation hook.
+	 */
+	public function insert_default_template_data() {
+		$current_theme_name = get_option( 'stylesheet' );
+		/**
+		 * This site option will be used to indicate that template data has already been
+		 * inserted for this theme, in order to prevent this functionality from running
+		 * more than once.
+		 */
+		$fse_template_data_option = $current_theme_name . '-fse-template-data';
+
+		if ( get_option( $fse_template_data_option ) ) {
+			/*
+			 * Bail here to prevent inserting the FSE data twice for any given theme.
+			 * Multiple themes will still be able to insert different template parts.
+			 */
+			return;
+		}
+
+		$this->register_template_post_types();
+
+		$header_id = wp_insert_post(
+			[
+				'post_title'     => 'Header',
+				'post_content'   => $this->get_header_content(),
+				'post_status'    => 'publish',
+				'post_type'      => 'wp_template_part',
+				'comment_status' => 'closed',
+				'ping_status'    => 'closed',
+			]
+		);
+
+		if ( ! term_exists( "$current_theme_name-header", 'wp_template_part_type' ) ) {
+			wp_insert_term( "$current_theme_name-header", 'wp_template_part_type' );
+		}
+
+		wp_set_object_terms( $header_id, "$current_theme_name-header", 'wp_template_part_type' );
+
+		$footer_id = wp_insert_post(
+			[
+				'post_title'     => 'Footer',
+				'post_content'   => $this->get_footer_content(),
+				'post_status'    => 'publish',
+				'post_type'      => 'wp_template_part',
+				'comment_status' => 'closed',
+				'ping_status'    => 'closed',
+			]
+		);
+
+		if ( ! term_exists( "$current_theme_name-footer", 'wp_template_part_type' ) ) {
+			wp_insert_term( "$current_theme_name-footer", 'wp_template_part_type' );
+		}
+
+		wp_set_object_terms( $footer_id, "$current_theme_name-footer", 'wp_template_part_type' );
+
+		$page_template_id = wp_insert_post(
+			[
+				'post_title'     => 'Page Template',
+				'post_content'   => $this->get_template_content( $header_id, $footer_id ),
+				'post_status'    => 'publish',
+				'post_type'      => 'wp_template',
+				'comment_status' => 'closed',
+				'ping_status'    => 'closed',
+			]
+		);
+
+		if ( ! term_exists( "$current_theme_name-page-template", 'wp_template_type' ) ) {
+			wp_insert_term( "$current_theme_name-page-template", 'wp_template_type' );
+		}
+
+		wp_set_object_terms( $page_template_id, "$current_theme_name-page-template", 'wp_template_type' );
+
+		add_option( $fse_template_data_option, true );
+	}
+
+	/**
+	 * Returns default header template part content.
+	 *
+	 * @return string
+	 */
+	public function get_header_content() {
```

Comment:
We talked about this in a call, but I think this data should be coming from an API.

Reply:
Agreed, that that would be a lot better than hardcoding it here. I don't see any blockers to doing that atm. 👍